egs2/TEMPLATE/asr1/scripts/feats/perform_kmeans.sh

```diff
@@ -0,0 +1,327 @@
+#!/bin/bash
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
+stage=1
+stop_stage=100
+skip_stages=
+python=python3       # Specify python to execute espnet commands.
+train_set=           # Name of training set
+dev_set=             # Name of valid set
+other_sets=           # Name of other sets
+datadir=dump/raw
+featdir=dump/hubert_feats
+km_dir=
+dictdir=
+alignment_phoneme_dir="data/mfa_phoneme_alignment"
+phn_sets="dev-other dev-clean"
```

Comment:
This seems too specific to Librispeech.

Reply:
Yeah, I intended to do so. Basically, the current quality measurement is specially designed for LibriSpeech. That's why the quality measure python script is under librispeech/{ssl1,asr2}/local. We need to update this part and the quality measurement script in the future. I'm afraid that the other dataset's alignment doesn't match the tsv format as used in librispeech.

Reply:
I added a comment in the Stage 4.